File: namastack-outbox-examples/namastack-outbox-example-java/src/main/java/io/namastack/demo/GenericOutboxHandler.java
package io.namastack.demo;

import io.namastack.outbox.handler.OutboxHandler;
import io.namastack.outbox.handler.OutboxRecordMetadata;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.stereotype.Component;

@Component
public class GenericOutboxHandler implements OutboxHandler {

  private static final Logger logger = LoggerFactory.getLogger(GenericOutboxHandler.class);

  @Override
  public void handle(Object payload, OutboxRecordMetadata metadata) {
    String className = payload.getClass().getSimpleName();
    logger.info("[Handler] Publish {}: {}", className, metadata.getKey());
    ExternalBroker.publish(payload, metadata.getKey());
  }
}

File: namastack-outbox-examples/namastack-outbox-example-tracing/src/main/kotlin/io/namastack/demo/GenericOutboxHandler.kt
package io.namastack.demo

import io.namastack.outbox.handler.OutboxHandler
import io.namastack.outbox.handler.OutboxRecordMetadata
import org.slf4j.LoggerFactory
import org.springframework.stereotype.Component

@Component
class GenericOutboxHandler(
    private val externalBroker: ExternalBroker
) : OutboxHandler {
    private val logger = LoggerFactory.getLogger(GenericOutboxHandler::class.java)

    override fun handle(
        payload: Any,
        metadata: OutboxRecordMetadata,
    ) {
        logger.info("[Handler] Publish {}: {}", payload::class.simpleName, metadata.key)
        externalBroker.publish(payload, metadata.key)
    }
}
